File: orchestrator/python/orchestrator/worker.py
import uuid
import queue

from orchestrator import task


class Worker:
    name: str
    queue: queue.Queue
    db: dict[uuid.UUID, task.Task]
    task_count: int

    def __init__(self, **kwargs):
        for k, v in kwargs.items():
            self.__setattr__(k, v)

    def collect_stats(self):
        print("I will collect stats")

    def run_task(self):
        print("I will start or stop a task")

    def start_task(self):
        print("I will start a task.")

    def stop_task(self):
        print("I will stop a task")

File: orchestrator/go/worker/worker.go
package worker

import (
	"github.com/golang-collections/collections/queue"
	"github.com/google/uuid"
	"github.com/supertylerc/sre-portfolio/orchestrator/go/task"
	"log/slog"
)

type Worker struct {
	Name      string
	Queue     queue.Queue
	Db        map[uuid.UUID]*task.Task
	TaskCount int
}

func (w *Worker) CollectStats() {
	slog.Debug("I will collect stats")
}

func (w *Worker) RunTask() {
	slog.Debug("I will start or stop a task")
}

func (w *Worker) StartTask() {
	slog.Debug("I will start a task.")
}

func (w *Worker) StopTask() {
	slog.Debug("I will stop a task")
}
